massa-models/src/amount.rs

```diff
@@ -33,21 +33,53 @@ impl Amount {
         Self(0)
     }
 
-    /// Create an Amount from the form `mantissa / (10^scale)`
-    /// Panics on any error.
-    /// Used for constant initialization.
+    /// Convert to decimal
+    fn to_decimal(&self) -> Decimal {
+        Decimal::from_u64(self.0)
+            .unwrap() // will never panic
+            .checked_div(AMOUNT_DECIMAL_FACTOR.into()) // will never panic
+            .unwrap() // will never panic
+    }
+
+    /// Create an Amount from a Decimal
+    fn from_decimal(dec: Decimal) -> Result<Self, ModelsError> {
+        let res = dec
+            .checked_mul(AMOUNT_DECIMAL_FACTOR.into())
+            .ok_or_else(|| ModelsError::AmountParseError("amount is too large".to_string()))?;
+        if res.is_sign_negative() {
+            return Err(ModelsError::AmountParseError(
+                "amounts cannot be strictly negative".to_string(),
+            ));
+        }
+        if !res.fract().is_zero() {
+            return Err(ModelsError::AmountParseError(format!(
+                "amounts cannot be more precise than 1/{}",
+                AMOUNT_DECIMAL_FACTOR
+            )));
+        }
+        let res = res.to_u64().ok_or_else(|| {
+            ModelsError::AmountParseError(
+                "amount is too large to be represented as u64".to_string(),
+            )
+        })?;
+        Ok(Amount(res))
+    }
+
+    /// Create an Amount from the form `mantissa / (10^scale)` in a const way.
+    /// WARNING: Panics on any error.
```

Comment:
I think we could use https://crates.io/crates/static_assertions in this function to replace both assert and panic and catch errors at compile time.
Maybe as a follow up or separated issue?

Reply:
I didn't want to go for more dependencies but it could be a nice followup

Reply:
Just for information I added https://github.com/massalabs/massa/issues/4095 to keep that in mind